src/libMessage/Messenger.cpp

```diff
@@ -302,894 +560,1935 @@ bool Messenger::GetNodeDSBlock(const vector<unsigned char>& src,
     return SerializeToArray(result, dst, offset);
 }
 
-bool Messenger::GetLookupSetNetworkIDFromSeed(const vector<unsigned char>& src,
-                                              const unsigned int offset,
-                                              string& networkID)
+bool Messenger::GetConsensusChallenge(
+    const vector<unsigned char>& src, const unsigned int offset,
+    const uint32_t consensusID, const vector<unsigned char>& blockHash,
+    const uint16_t leaderID, CommitPoint& aggregatedCommit,
+    PubKey& aggregatedKey, Challenge& challenge, const PubKey& leaderKey)
 {
     LOG_MARKER();
 
-    LookupSetNetworkIDFromSeed result;
+    ConsensusChallenge result;
 
     result.ParseFromArray(src.data() + offset, src.size() - offset);
 
     if (!result.IsInitialized())
+    {
+        LOG_GENERAL(WARNING, "ConsensusChallenge initialization failed.");
+        return false;
+    }
+
+    if (result.consensusinfo().consensusid() != consensusID)
     {
         LOG_GENERAL(WARNING,
-                    "LookupSetNetworkIDFromSeed initialization failed.");
+                    "Consensus ID mismatch. Expected: "
+                        << consensusID
+                        << " Actual: " << result.consensusinfo().consensusid());
         return false;
     }
 
-    networkID = result.networkid();
+    if ((result.consensusinfo().blockhash().size() != blockHash.size())
+        || !equal(blockHash.begin(), blockHash.end(),
+                  result.consensusinfo().blockhash().begin()))
+    {
+        LOG_GENERAL(WARNING, "Block hash mismatch.");
+        return false;
+    }
 
-    return true;
-}
+    if (result.consensusinfo().leaderid() != leaderID)
+    {
+        LOG_GENERAL(WARNING,
+                    "Leader ID mismatch. Expected: "
+                        << leaderID
+                        << " Actual: " << result.consensusinfo().leaderid());
+        return false;
+    }
 
-bool Messenger::SetLookupGetStateFromSeed(vector<unsigned char>& dst,
-                                          const unsigned int offset,
-                                          const uint32_t listenPort)
-{
-    LOG_MARKER();
+    ProtobufByteArrayToSerializable(result.consensusinfo().aggregatedcommit(),
+                                    aggregatedCommit);
+    ProtobufByteArrayToSerializable(result.consensusinfo().aggregatedkey(),
+                                    aggregatedKey);
+    ProtobufByteArrayToSerializable(result.consensusinfo().challenge(),
+                                    challenge);
 
-    LookupGetStateFromSeed result;
+    vector<unsigned char> tmp(result.consensusinfo().ByteSize());
+    result.consensusinfo().SerializeToArray(tmp.data(), tmp.size());
 
-    result.set_listenport(listenPort);
+    Signature signature;
 
-    if (!result.IsInitialized())
+    ProtobufByteArrayToSerializable(result.signature(), signature);
+
+    if (!Schnorr::GetInstance().Verify(tmp, signature, leaderKey))
     {
-        LOG_GENER
```

Comment:
How about using for(const auto i : result.consensusinfo().bitmap())?

Reply:
ok